charts/argo-templates/templates/diff-bq-table.yaml

```diff
@@ -137,6 +137,29 @@ spec:
           - name: rows-to-append-count
             valueFrom:
               parameter: '{{ "{{tasks.query-row-count.outputs.result}}" }}'
+      metrics:
+        prometheus:
+          - name: success-counter
+            help: 'Number of successful runs'
+            when: '{{status}} == Succeeded'
+            counter: 
+              value: '1'
+          - name: failure-counter
+            help: 'Number of failed runs'
+            when: '{{status}} == Failed'
+            counter: 
+              value: '1'
+          - name: rows-to-delete-gauge
+            help: 'Number of rows to be deleted'
+            when: '{{status}} == Succeeded'
+            gauge: 
+              value: '{{ outputs.parameters.ids-to-delete-count }}'
+          - name: rows-to-append-gauge
+            help: 'Number of rows to be added'
+            when: '{{status}} == Succeeded'
+            gauge: 
+              value: '{{ outputs.parameters.rows-to-append-count }}'
```

Comment:
It might be more idiomatic Prometheus if we had a single "row-count" metric and used labels to distinguish between appends and deletes, WDYT? I haven't read through all the linked docs about cardinality explosion yet to know which layout is more efficient.

Reply:
I definitely agree that it would be more efficient that way but would have to think about how to implement that since the counts come from different variables. And from what I read, cardinality explosions mostly become an issue when you have 100s of different label values and 100s of replicas of whatever service you're running.

Either way, it might make sense to label the values by table as well, what do you think?

Reply:
Labeling by table makes sense to me!